ste/mgr-JobPartMgr.go

```diff
@@ -80,14 +78,11 @@ type IJobPartMgr interface {
 // number of available network sockets on resource-constrained Linux systems. (E.g. when
 // 'ulimit -Hn' is low).
 func NewAzcopyHTTPClient(maxIdleConns int) *http.Client {
+	const concurrentDialsPerCpu = 10 // exact value doesn't matter too much, but too low will be too slow, and too high will reduce the beneficial effect on thread count
 	return &http.Client{
 		Transport: &http.Transport{
-			Proxy: common.GlobalProxyLookup,
-			DialContext: newDialRateLimiter(&net.Dialer{
-				Timeout:   30 * time.Second,
-				KeepAlive: 30 * time.Second,
-				DualStack: true,
-			}).DialContext,
+			Proxy:                  common.GlobalProxyLookup,
+			MaxConnsPerHost:        concurrentDialsPerCpu * runtime.NumCPU(),
```

Comment:
Not 100% sure, but `MaxConnsPerHost` seems like the perfect fit for `AZCOPY_CONCURRENCY_VALUE` rather than having a hardcoded value.

Reply:
We do not want to change our current behavior drastically as it may break some of the running scenarios. Changes proposed in the PR appears to be a safer choice to me.